src/Data/Repositories/Repository.cs

```diff
@@ -18,115 +15,38 @@ protected Repository(ApplicationContext context)
-
-            query = BaseQuery(query, queryParameters);
+            var createdEntity = await Set.AddAsync(entity);
 
-            if (queryParameters.PaginationRule != null)
-                query = PaginationQuery(query, queryParameters.PaginationRule);
-
-            return await query.ToListAsync();
+            return createdEntity.Entity;
         }
 
-        public virtual async Task<TEntity> CreateAsync(TEntity entity)
+        public TEntity Update(TEntity entity)
         {
-            var result = await Set.AddAsync(entity);
-            return result.Entity;
-        }
+            var updatedEntity = Set.Update(entity);
 
-        public virtual TEntity Update(TEntity entity)
-        {
-            var result = Set.Update(entity);
-            return result.Entity;
+            return updatedEntity.Entity;
         }
 
-        public virtual async Task<TEntity> DeleteAsync(TKey id)
+        public async Task<TEntity> DeleteAsync(TKey id)
         {
             var entityToDelete = await Set.FindAsync(id);
 
             if (entityToDelete != null)
-            {
-                Set.Remove(entityToDelete);
-            }
-
-            return entityToDelete;
-        }
-
-        protected virtual IQueryable<TEntity> BaseQuery(IQueryable<TEntity> queryable, QueryParameters<TEntity, TKey> queryParameters)
-        {
-            if (queryParameters.FilterRule?.FilterExpression != null)
-                queryable = FilterQuery(queryable, queryParameters.FilterRule);
-
-            if (queryParameters.SortRule?.SortExpression != null)
-                queryable = SortQuery(queryable, queryParameters.SortRule);
-
-            return queryable;
-        }
-
-        protected virtual IQueryable<TEntity> FilterQuery(IQueryable<TEntity> queryable, FilterRule<TEntity, TKey> filterRule)
-        {
-            queryable = queryable.Where(filterRule.FilterExpression);
-
-            return queryable;
-        }
-
-        protected virtual IQueryable<TEntity> SortQuery(IQueryable<TEntity> queryable, SortRule<TEntity, TKey> sortRule)
-        {
-            queryable = sortRule.InAscendingOrder
-                    ? queryable.OrderBy(sortRule.SortExpression)
-                    : queryable.OrderByDescending(sortRule.SortExpression);
-
-            return queryable;
-        }
-
-        protected virtual IQueryable<TEntity> PaginationQuery(IQueryable<TEntity> queryable, PaginationRule paginationRule)
-        {
-            if (paginationRule == null)
-                throw new ArgumentException("Pagination rule can't be null.", nameof(paginationRule));
-
-            if (paginationRule.Index < 0 || paginationRule.Size <= 0)
-                throw new ArgumentException("Pagination rule is not valid.", nameof(paginationRule));
-
-            queryable = queryable.Skip(paginationRule.Index * paginationRule.Size).Take(paginationRule.Size);
-
-            return queryable;
+                return Set.Remove(entityToDelete).Entity;
```

Comment:
I think, that we can use just Remove method

Reply:
Done